PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStoragePanel.kt

```diff
@@ -195,15 +196,31 @@ open class SparkSubmissionJobUploadStoragePanel: JPanel(), Disposable, ILogger {
                 AzureStorageVirtualFileSystem.VFSSupportStorageType.ADLSGen2 -> {
                     // for issue #3159, upload path maybe not ready if switching cluster fast so path is the last cluster's path
                     // if switching between gen2 clusters, need to check account is matched
-                    val isPathValid = uploadRootPath?.matches(SparkBatchJob.AdlsGen2RestfulPathPattern.toRegex())
-                            ?: false
-                    val isAccountMatch = uploadRootPath?.contains(account ?: "") ?: false
-                    if (!isPathValid || !isAccountMatch || StringUtils.isBlank(account) || StringUtils.isBlank(accessKey)) {
+                    if (uploadRootPathUri.accountName != account) {
                         return null
                     }
 
-                    fileSystem = ADLSGen2FileSystem(SharedKeyHttpObservable(account, accessKey), uploadRootPath)
-                    return AdlsGen2VirtualFile(fileSystem.root, true, fileSystem)
+                    // Prepare HttpObservable for different cluster type
+                    val http =
+                        if (cluster is ArcadiaSparkCompute) {
```

Comment:
May also for HIB ESP cluster? We might have to add an interface or put those default storage logic into `ClusterDetail`